python/lsst/daf/butler/core/storedFileInfo.py

```
@@ -19,87 +19,51 @@
-        self._storageClass = storageClass
-        assert checksum is None or isinstance(checksum, str)
-        self._checksum = checksum
-        assert size is None or isinstance(size, int)
-        self._size = size
 
-    @property
-    def formatter(self):
-        """Full name of formatter (`str`).
-        """
-        return self._formatter
+@dataclass
+class StoredFileInfo(StoredDatastoreItemInfo):
+    """Datastore-private metadata associated with a file stored in a Datastore.
+    """
+    formatter: str
+    """Fully-qualified name of Formatter."""
 
-    @property
-    def path(self):
-        """Path to Dataset (`str`).
-        """
-        return self._path
+    path: str
+    """Path to dataset within Datastore."""
 
-    @property
-    def storageClass(self):
-        """StorageClass used (`StorageClass`).
-        """
-        return self._storageClass
+    storageClass: StorageClass
+    """StorageClass associated with Dataset."""
 
-    @property
-    def checksum(self):
-        """Checksum (`str`).
-        """
-        return self._checksum
+    checksum: Optional[str]
+    """Checksum of the serialized dataset."""
 
-    @property
-    def size(self):
-        """Size of stored object in bytes (`int`).
-        """
-        return self._size
+    file_size: int
+    """Size of the serialized dataset in bytes."""
 
-    def __repr__(self):
-        return f'{type(self).__qualname__}(path="{self.path}", formatter="{self.formatter}"' \
-            f' size={self.size}, checksum="{self.checksum}", storageClass="{self.storageClass.name}")'
+    def __post_init__(self):
+        # This modification prevents the data class from being frozen.
+        if isinstance(self.formatter, str):
+            # We trust that this string refers to a Formatter
+            pass
+        elif isinstance(self.formatter, Formatter) or \
+                (inspect.isclass(self.formatter) and issubclass(self.formatter, Formatter)):
```

Why do we need the `inspect` check?  Does `issubclass` raise instead of returning `False` when given non-types?

---

Yes. Amazingly enough `issubclass` raises if it's given anything other than a class, rather than returning `False` (or checking whether the instance is equivalent).